src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -327,6 +327,13 @@ internal static SafeSslHandle AllocateSslHandle(SafeFreeSslCredentials credentia
                         Crypto.ErrClearError();
                     }
 
+                    // We allow post-handshake client authentication with TLS1.3 in the same
+                    // scenarios when client wants to support renegotiation in older TLS versions
+                    if (sslAuthenticationOptions.AllowRenegotiation)
```

Comment:
For older TLS versions it  would work with false, right? (since we effectively ignore AllowRenegotiation on Linux. 
My original though was to check for presence of certificate or certificate selection callback. 
We may also consider check if the platform _can_ support Tls13.

Reply:
Replaced by checking if user provided cert (or cert callback)